File: native/mixdog-graph/tests/fixtures/parity/dart/sample.dart
// Parity fixture: declaration and import shapes the graph reports for Dart.
import 'dart:async';
import 'package:meta/meta.dart';
import './helper.dart';
export 'src/api.dart';
part 'sample.g.dart';

enum Mode { fast, slow }

mixin Loggable {
  void log(String message) => print(message);
}

extension StoreExtras on Store {
  String get label => 'store:${name}';
}

class Store with Loggable {
  Store(this.name);

  final String name;

  String read(String key) {
    String nested(String value) => value.trim();
    return nested('$name:$key');
  }

  String get upper => name.toUpperCase();

  set rename(String value) => log(value);

  bool operator ==(Object other) => other is Store && other.name == name;

  @override
  int get hashCode => name.hashCode;
}

Future<Store> build(String name) async => Store(name);

File: native/mixdog-graph/tests/fixtures/parity/zig/sample.zig
// Parity fixture: declaration and import shapes the graph reports for Zig.
const std = @import("std");
const helper = @import("./helper.zig");

pub const Store = struct {
    name: []const u8,
    size: usize = 0,

    pub fn read(self: Store, key: []const u8) usize {
        return self.name.len + key.len;
    }
};

pub const Mode = enum {
    fast,
    slow,
};

const Payload = union(Mode) {
    fast: u32,
    slow: u64,
};

pub fn build(name: []const u8) Store {
    return Store{ .name = name };
}

fn internal(value: usize) usize {
    return value + helper.offset;
}

pub fn main() !void {
    const store = build("main");
    std.debug.print("{d}\n", .{internal(store.read("k"))});
}
